frontend/src/store/index.ts

```diff
@@ -1,20 +1,59 @@
 import Vue from "vue";
 import Vuex from "vuex";
+import axios from "axios";
+import { AxiosInstance } from "axios";
 
 Vue.use(Vuex);
 
+interface State {
+  isAuthenticated: boolean;
+  username: string | undefined;
+  avatar: string | undefined | null;
+  id: string | undefined;
+  token: string | undefined;
+}
+
+const state: State = {
+  isAuthenticated: false,
+  username: undefined,
+  avatar: null,
+  id: undefined,
+  token: undefined,
+};
+
```

Comment:
Yes, it will probably break the chat. The reasoning behind it is that one needed the `id` of the current user to make HTTP requests. I could add `id` to the loginDto. it does not seem to be a problem as there are getters for variables in the store, but I would add a type for this object. What do you think?

Reply:
Yes, I marked it as breaking in the commit name. I will try and not change the way one can use the store. Can you point me to the files where you use the store in your Chat please?

Reply:
I agree this is breaking (hence the `!` in the commit message). I could change the interface of the store to something like:
```typescript
interface User {
  username: string;
  id: string;
  token: string;
  avatar: string | undefined | null;
}

interface State {
  isAuthenticated: boolean;
  user: User | undefined;
}```

I could still have getters for the `username`, `id`, etc. What do you think?

Reply:
Ah, apologies! I didn't see `!`.

I think we should keep your changes as they are. The other conflicts aren't merged yet, so I can make changes there